source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -1342,6 +1236,125 @@ bool COpenGLDriver::beginScene(bool backBuffer, bool zBuffer, SColor color,
 }
 
 
+core::smart_refctd_ptr<IGPUShader> COpenGLDriver::createGPUShader(const asset::ICPUShader* _cpushader)
+{
+    return core::make_smart_refctd_ptr<COpenGLShader>(_cpushader);
+}
+
+core::smart_refctd_ptr<IGPUSpecializedShader> COpenGLDriver::createGPUSpecializedShader(const IGPUShader* _unspecialized, const asset::ISpecializationInfo* _specInfo)
+{
+    const COpenGLShader* glUnspec = static_cast<const COpenGLShader*>(_unspecialized);
+    const asset::ICPUShader* cpuUnspec = glUnspec->getCPUCounterpart();
+
+    const std::string& EP = _specInfo->entryPoint;
+    const asset::E_SHADER_STAGE stage = _specInfo->shaderStage;
+
+    const asset::ICPUShader* spvCPUShader = nullptr;
+    if (cpuUnspec->containsGLSL()) {
+        //TODO insert enabled extensions #defines into GLSL
```

Comment:
[epic.txt](https://github.com/buildaworldnet/IrrlichtBAW/files/3708060/epic.txt)
[epicGLSL.txt](https://github.com/buildaworldnet/IrrlichtBAW/files/3708061/epicGLSL.txt)
List of extensions you can check.

You can add `virtual const std::string* getSupportedGLSLExtensionsBegin() const = 0;` and  `virtual const std::string* getSupportedGLSLExtensionsEnd() const = 0;` methods to `IVideoCapabilityReporter` that will give you the lists of `GL_` extensions supported (beware different on OpenGL and Vulkan, also `SPV_` extensions don't actually have a `SPV_` prefix for GLSL source being compiled, they are `GL_`)

Then `createSPIRVFromGLSL` could take a pair of iterators for the extension strings to stick in as defines

Reply:
@Crisspl TODO.